### PR TITLE
Replace some jenkins jobs with a GH Actions

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

**Review by krionbsd — DISMISSED** [review]

**Comment by s0undt3ch** [comment]
GH Actions switched to docker containers

**Review by bryceml — DISMISSED** [review]

**Review by bryceml — DISMISSED** [review]

**Review by krionbsd — DISMISSED** [review]

**Review by krionbsd — DISMISSED** [review]

**Review by ScriptAutomate — COMMENTED** [review]
This is so awesome that we are migrating stuff to GitHub Actions! It's great that I can see what these actions are doing via the **Check** tab of the PR:
- https://github.com/saltstack/salt/pull/58983/checks

The only confirmation I'd like make is about this section where a **Good**, **Bad**, **Good** example set exists, though I think it is intended to have only one **Good** and one **Bad**? Just want to confirm whether it is intentional:

https://github.com/saltstack/salt/pull/58983/files?file-filters%5B%5D=.py&file-filters%5B%5D=.rst#diff-595e29dbd27a244b4837db52d7f5ffbd4dd352c4e7605b79a12921f3a122aad2R59-R80

**Comment by s0undt3ch** [comment]
I can make it just one Good and one Bad, two good and one Bad was intentional since they refer to different things.
But we can join examples

File: doc/topics/development/conventions/style.rst [hunk 1]
```diff
@@ -66,15 +56,23 @@ Formatting Strings
...
+    log.debug("%s and then some", data)
 
-Make sure to use indices or identifiers in the format brackets, since empty
-brackets are not supported by python 2.6.
+Bad:
+
+.. code-block:: python
+
+    data = "some text"
+    log.debug("{} and then some".format(data))
```
Comment (ScriptAutomate) [inline doc/topics/development/conventions/style.rst:74]: Should we create an issue for where this is done throughout Salt?

```
# Shows a lot of instances of use
grep -rn "log.debug(" | grep "\.format"
```

```
$ grep -r "log.debug(" | grep "\.format" | wc -l
97

$ grep -r "log.debug(" | grep "\.format" | grep -v "tests/" | wc -l
72
```

Comment (s0undt3ch) [reply doc/topics/development/conventions/style.rst:74]: There's a disabled PyLint rule which will take care of that.
The issue with enabling that is fixing the tons of bad usage across the salt code base.
We could consider that a stretch goal for Aluminum though.

**Review by ScriptAutomate — DISMISSED** [review]

File: doc/topics/development/conventions/style.rst (hunk 1 above)
Comment (s0undt3ch) [reply doc/topics/development/conventions/style.rst:74]: FYI, https://github.com/saltstack/salt/pull/59031

**Review by dwoz — APPROVED** [review]

**Pull request merged**